core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -1151,6 +1150,37 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext with Eventu
     assert(sc.listJars().exists(_.contains("org.apache.hive_hive-storage-api-2.7.0.jar")))
     assert(sc.listJars().exists(_.contains("commons-lang_commons-lang-2.6.jar")))
   }
+
+  test("SPARK-34346: hadoop configuration priority for spark/hive/hadoop configs") {
```

Comment:
For good measure, do we want to also test that the default io.file.buffer.size you get from a plain Spark configuration is in fact 65536?

Reply:
I guess that will cause test flakiness if it gets a nondefault value somewhere due to the non-deterministic test order in CIs. I did a pre-check for loading hive-site.xml explicitly to ensure it gets loaded but not pollute the final result.

Reply:
oh, I can just set it explicitly too